File: java/src/Primes.java
import java.util.*;

public class Primes {

    public static void main(String[] args) {
        int attempts = Integer.parseInt(args[0]);
        int count = Integer.parseInt(args[1]);

        Date time_0 = new Date();
        for (int attempt = 0; attempt < attempts; attempt++) {
            print_primes(100_000_000, count);
        }
        Date time_1 = new Date();

        double duration = (time_1.getTime() - time_0.getTime()) / 1000.0;
        System.out.println("---");
        System.out.println(String.format("Attempts: %s, duration: %s sec", attempts, duration));
    }

    public static void print_primes(int start_from, int count) {
        System.out.println("Hello, print_primes!");

        int n = start_from;
        while (count > 0) {
            Date time_0 = new Date();
            boolean is_prime = is_prime_number(n);
            Date time_1 = new Date();

            if (is_prime) {
                long duration = (time_1.getTime() - time_0.getTime());
                System.out.println(n + ", " + duration + " msec");
                count = count - 1;
            }

            n = n + 1;
        }
    }

    public static boolean is_prime_number(int number) {
        int i = 2;
        while (i < number) {
            if (number % i == 0) {
                return false;
            }
            i = i + 1;
        }

        return true;
    }

}

File: python/primes.py
import datetime
import sys


def is_prime_number(number):
    i = 2
    while i < number:
        if number % i == 0:
            return False
        i = i + 1

    return True


def print_primes(from_number, count):
    print("Hello, print_primes!")

    n = from_number
    while count > 0:
        time_0 = datetime.datetime.now()
        is_prime = is_prime_number(n)
        time_1 = datetime.datetime.now()

        if is_prime:
            duration = int(diff_msec(time_1, time_0))
            print(str(n) + ", " + str(duration) + " msec")
            count = count - 1

        n = n + 1


def diff_msec(dt_2, dt_1):
    done_msec = (dt_2 - dt_1).days * 24 * 60 * 60 * 1000 + (dt_2 - dt_1).seconds * 1000 + (dt_2 - dt_1).microseconds / 1000
    return done_msec


def main():
    attempts = int(sys.argv[1])
    count = int(sys.argv[2])

    time_0 = datetime.datetime.now()
    for attempt in range(attempts):
        print_primes(100_000_000, count)
    time_1 = datetime.datetime.now()

    duration = int(diff_msec(time_1, time_0)) / 1000
    print("---")
    print("Attempts: " + str(attempts) + ", duration: " + str(duration) + " sec")


###
main()
